### ISSUE TITLE
Directories with `{}` characters display strangely

### ISSUE THREAD

**Issue by jsdevel** [scrooloose/nerdtree#555]
![nerdtree](https://cloud.githubusercontent.com/assets/2605980/13368349/d1839500-dca6-11e5-8a04-eca105a23b9f.png)

**Comment by lifecrisis** [comment]
Confirmed bug. Has to do with highlight regular expression confusing the directory name with a bookmark flag.

**Comment by PhilRunninger** [comment]
@jsdevel , would you be willing to test the pull request #868 I've put together for this issue? Details are in the PR. Thanks.

**Comment by jsdevel** [comment]
@PhilRunninger sure!  may take me a couple days though

**Comment by jsdevel** [comment]
@PhilRunninger I checked out that branch.  Doesn't seem to have fixed it.

![not-working](https://user-images.githubusercontent.com/2605980/44276487-7580b580-a1fc-11e8-98bd-0139bae2ad83.png)

**Comment by jsdevel** [comment]
Why is it showing . and .. directories?

**Comment by PhilRunninger** [comment]
I agree; the . and .. directories shouldn't be displayed. I couldn't find a way to reproduce this though. What environment are you in? OS? Shell? Can you share the complete directory listing with as much detail as possible as it appears in the shell? Does this happen in all NERTree directories, or just {foo}?

**Comment by jsdevel** [comment]
Fedora Linux, Terminator, happens any time Tuesday directory has curlies.

**Comment by PhilRunninger** [comment]
What version of Vim are you using? There have been several changes to the `glob()` and `globpath()` functions over the years (these functions are used to get a directories children), and I wonder if your particular version has not been taken into account. Please include the whole content of the `:version` command.

**Comment by PhilRunninger** [comment]
I believe I've figured it where it's happening, but not necessarily why. Using this example,
```
$ mkdir /tmp/{foo}
$ touch /tmp/{foo}/.bar
```
Vim exhibits the following behavior
```vim
:echo globpath('/tmp/{foo}', '.*')
/tmp/{foo}/./
/tmp/{foo}/../
/tmp/{foo}/.bar
:echo globpath('/tmp/\{foo\}', '.*')
/tmp/{foo}/.
/tmp/{foo}/..
/tmp/{foo}/.bar
```
Notice the difference between `{foo}` and `\{foo\}`. I'm not sure what the special meaning of the curly braces is inside the `globpath()` function; I haven't been able to find anything in the help pages or on Google.

Regardless, it seems there is code after this in NERDTree that tries to deal with this situation when eliminating the `.` and `..` entries, but it wasn't working correctly. Using `fnamemodify()` it first looks at the `:t`ail of the string. With a trailing slash, an empty string is returned. If that happens, it then used `:t:t`, which doesn't work. That would be getting the tail of a tail, which is still an empty string.

Changing that to `:h:t` works correctly, getting the `:h`ead of the path string (minus the final slash), and then the `:t`ail of that. It accurately finds `.` and `..`, and can then remove them. 

I updated the pull request. Give it another try, and let me know if it works for you.
